frontend/packages/ux-editor/src/components/Elements/ExportForm.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { StudioButton } from '@studio/components';
+import { useAppContext } from '../../hooks';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useFormLayoutsQuery } from '../../hooks/queries/useFormLayoutsQuery';
+import { useOptionListsQuery } from '../../hooks/queries/useOptionListsQuery';
+import { useTextResourcesQuery } from 'app-shared/hooks/queries';
+import { generateExportFormFormat } from '../../utils/exportUtils';
+import { useFormLayoutSettingsQuery } from '@altinn/ux-editor/hooks/queries/useFormLayoutSettingsQuery';
+
+export const ExportForm = () => {
+  const { org, app } = useStudioEnvironmentParams();
+  const { selectedFormLayoutSetName } = useAppContext();
+  const { data: formLayouts } = useFormLayoutsQuery(org, app, selectedFormLayoutSetName);
+  const { data: optionLists } = useOptionListsQuery(org, app);
+
+  const { data: textResources } = useTextResourcesQuery(org, app);
+  const { data: settings } = useFormLayoutSettingsQuery(org, app, selectedFormLayoutSetName);
+
+  const handleExportClick = () => {
+    if (formLayouts && textResources) {
+      const exportFormat = generateExportFormFormat(
+        settings?.pages?.order,
+        formLayouts,
+        selectedFormLayoutSetName,
+        app,
+        textResources,
+        optionLists,
+        'nb',
+        false,
+      );
+      const blob = new Blob([JSON.stringify(exportFormat)], { type: 'application/json' });
```

Comment:
What about creating a class that is reusable to generate blobs to download? 😊 I think the implementation could be something like this:

```typescript
export class BlobDownloader<Data> {
  private readonly data: Data;
  private readonly fileType: string;

  constructor(data: Data, fileType = "application/json") {
    this.data = data;
    this.fileType = fileType;
  }

  public getDownloadURL(): string {
    const blob = this.generateBlobToDownlaod();
    return this.generateDownloadUrl(blob);
  }

  public revokeDownloadURL(url: string): void {
    return URL.revokeObjectURL(url);
  }

  private generateBlobToDownlaod(): Blob {
    return new Blob([JSON.stringify(this.data)], { type: this.fileType });
  }

  private generateDownloadUrl(blob: Blob): string {
    return URL.createObjectURL(blob);
  }
}
```

Then we could make a `StudioBlobDownloader` component that uses this class to generate the blob and ensure to revoke the download URL. 

```tsx
type StudioBlobDownloader<T> = {
    data: T;
    filename: string;
    fileType?: string;
    linkText: string;
}


export const StudioBlobDownloader = <T>({ data, filename, fileType, linkText }: StudioBlobDownloader<T>) => {
  const blobDownloader = new BlobDownloader<T>(data, fileType);
  const downloadUrl = blobDownloader.getDownloadURL();

  const handleRevoke = (): void => {
    blobDownloader.revokeDownloadURL(downloadUrl)
  }
      
  return (
    <Link
      href={downloadUrl}
      onClick={handleRevoke}
      download={fileName}
    >
      {linkText}
    </Link>
  );
};
```

Reply:
Great suggestions, thanks! 😊 
I have pushed some changes now, feel free to take a look. Some changes I made to your suggestion:
- The `BlobDownloader` class in practice only supports JSON, because the `generateBlobToDownload` method used `JSON.stringify...`. I have decided to use `string` for the data input instead to make the class generic. That way, if we have other data types we can pass them in as strings, and still use the downloader. In the future we might implement some specialized classes, f.ex. `JsonBlobDownloader` or `XmlBlobDownloader` that use `BlobDownloader` as a base class, and implement their own Blob generation. But I think we can leave that until we actually get a need for it.
- Calling the `revokeDownloadUrl` method `onClick` will not work, as it revokes the url _before_ the user is able to get to the download. An alternative would be to handle it in the state of the `StudioBlobDownloader` component, but I worry that that would result in code that is difficult to work with and not robust.
  Working with the DOM and setting up the link-element programatically seems to be an accepted way of doing this, f.ex. https://id.javascript.info/blob suggests this. However, I have moved that complexity into the `BlobDownloader` class as a separate method, to hide the complexity when we actually use it.